cli/daemon/run/infra/infra.go

```diff
@@ -223,7 +226,12 @@ func isDockerRunning(ctx context.Context) bool {
 // UpdateConfig updates the given config with infrastructure information.
 // Note that all the requisite services must have started up already,
 // which in practice means that (*optracker.AsyncBuildJobs).Wait must have returned first.
-func (rm *ResourceManager) UpdateConfig(cfg *config.Runtime, md *meta.Data, dbProxyPort int) {
+func (rm *ResourceManager) UpdateConfig(cfg *config.Runtime, md *meta.Data, dbProxyPort int) error {
+	useLocalEncoreCloudAPI, err := rm.setTestEncoreCloud(cfg)
```

Comment:
Originally I had it panic, but I changed it because the deamon would be the thing which panics, which would not give the error visibly to the CLI tool.

Given that the only call site of `UpdateConfig` already had error handling which passed the errors to the CLI, this felt less error prone in terms of us using it and not using it 100% correctly and seeing a random panic which might take time to debug - vs just immediately seeing a useful error